File: lib/Logger.cpp
// Copyright (c) 2014-2015 Josh Blum
//                    2022 Nicholas Corgan
// SPDX-License-Identifier: BSL-1.0

#include <SoapySDR/Logger.hpp>

void SoapySDR::log(const LogLevel logLevel, const std::string &message)
{
    return SoapySDR_log(logLevel, message.c_str());
}

void SoapySDR::vlogf(const SoapySDRLogLevel logLevel, const char *format, va_list argList)
{
    return SoapySDR_vlogf(logLevel, format, argList);
}

void SoapySDR::registerLogHandler(const LogHandler &handler)
{
    return SoapySDR_registerLogHandler(handler);
}

void SoapySDR::setLogLevel(const LogLevel logLevel)
{
    SoapySDR_setLogLevel(logLevel);
}

SoapySDR::LogLevel SoapySDR::getLogLevel(void)
{
    return SoapySDR_getLogLevel();
}

File: swig/java/maven/src/main/java/Pothosware/SoapySDR/Logger.java
// Copyright (c) 2022 Nicholas Corgan
// SPDX-License-Identifier: BSL-1.0

package Pothosware.SoapySDR;

public class Logger
{
    public interface SoapyLogger
    {
        void log(LogLevel logLevel, String message);
    }

    private static class LogHandler extends LogHandlerBase
    {
        public LogHandler(SoapyLogger _logger)
        {
            super();
            logger = _logger;
        }

        @Override
        public void handle(LogLevel logLevel, String message)
        {
            logger.log(logLevel, message);
        }

        private void setLogger(SoapyLogger _logger)
        {
            logger = _logger;
        }

        private SoapyLogger logger = null;
    }

    private static LogHandler logHandler = null;

    public static void log(LogLevel logLevel, String message)
    {
        LogHandlerBase.log(logLevel, message);
    }

    public static void logf(LogLevel logLevel, String format, Object... args)
    {
        log(logLevel, String.format(format, args));
    }

    public static void registerLogHandler(SoapyLogger logger)
    {
        logHandler = (logger != null) ? new LogHandler(logger) : null;
    }

    public static void setLogLevel(LogLevel logLevel)
    {
        LogHandlerBase.setLogLevel(logLevel);
    }
}
